src/main/osd/osd_elements.c

```diff
@@ -922,6 +922,17 @@ static void osdElementGpsSpeed(osdElementParms_t *element)
 {
     tfp_sprintf(element->buff, "%c%3d%c", SYM_SPEED, osdGetSpeedToSelectedUnit(gpsConfig()->gps_use_3d_speed ? gpsSol.speed3d : gpsSol.groundSpeed), osdGetSpeedToSelectedUnitSymbol());
 }
+
+static void osdElementEfficiency(osdElementParms_t *element)
+{
+    if (sensors(SENSOR_GPS) && ARMING_FLAG(ARMED) && STATE(GPS_FIX)) {
+        const int speed = osdGetSpeedToSelectedUnit(gpsSol.groundSpeed);
+        const int efficiency = speed > 0 ? getAmperage() * 10 / speed : 0;
+        const char unitSymbol = osdConfig()->units == OSD_UNIT_IMPERIAL ? SYM_MILES : SYM_KM;
+
+        tfp_sprintf(element->buff, "%4d%c/%c", abs(efficiency), SYM_MAH, unitSymbol);
```

Comment:
Note that the `%4` formatting doesn't limit to 4 digits. I think you need to think about some reasonableness limits on what's displayed (in conjunction about the lower speed threshold). For example it's obviously easy to draw more than 9.999 amps and if speed was slow this would overflow the display. I'm not even sure the instantaneous amperage is what you want. Perhaps you should be looking at consumption over an interval vs. speed in that interval.

Reply:
I've added a check that the value is within range and display `----` if it's not.

Perhaps you're right that instantaneous isn't ideal.  I could run it through a filter I suppose.  Or simply calculate once per second instead of every frame. hmmmm

Reply:
I played with the `ibat_lpf_period` setting, and found that cranking that up gave me exactly the result I wanted, resulting in "smoother" OSD values for both current and efficiency.  I don't think additional complication is needed in this feature, but am open to suggestions.

Do you have anything in particular in mind for the interval for "consumption over an interval" ?

Reply:
With a more filtered current meter (higher `ibat_lpf_period`) it looks okay. It can always be improved at some point if needed. For example using this method to calculate remaining range wouldn't work out well as the consumption is still too erratic.